### DISCUSSION TITLE
Custom layout not visible in settings

### DISCUSSION THREAD

**Discussion by MarkusLanger** [microsoft-search/pnp-modern-search#2205] [Q&A] [answered]
**Version used**
4.7.0

**Describe the bug**
I created a library component for a custom layout. I followed these steps => https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_layout/

But I can't the see custom layout in the settings.

**To Reproduce**
1. Clone repo https://github.com/MarkusLanger/pnp-search-custom-layout
2. npm install
3. gulp bundle --ship
4. gulp package-solution --ship
5. Deploy package in global App Catalog and deploy tenant wide
6. Configure custom library in  PnP Modern Search Result Webpart settings
7. Check Result Webpart layouts
8. No custom layout available

**Expected behavior**
Show custom layout in the settings

**Screenshots**
![image](https://user-images.githubusercontent.com/12131918/177333692-967dab8a-cfc8-4756-8c3b-15ee9ed78925.png)

![image](https://user-images.githubusercontent.com/12131918/177333534-1ea0f2c4-79cf-4be4-b4aa-8ddaae09a7b8.png)

![image](https://user-images.githubusercontent.com/12131918/177333728-4bd77974-c733-447f-89f8-ece8778417c3.png)

![image](https://user-images.githubusercontent.com/12131918/177333769-9c8ebf71-5079-434e-8f93-ab904b9aacc4.png)

**Desktop (please complete the following information):**

**Additional context**

####### Update 2022-07-06 #######
It works fine with PnP Modern Search version 4.6.1
![image](https://user-images.githubusercontent.com/12131918/177482400-b0907fba-f424-44c2-b27d-8f84a5b63f56.png)

![image](https://user-images.githubusercontent.com/12131918/177482436-83745238-c180-4197-8f3a-16bc30365bd4.png)

**Comment by FranckyC** [comment, marked as answer]
@MarkusLanger by any chance, did you check the 'Make this solution available to all sites in the organization' flag when deployed your library into the tenant app catalog?

**Reply by MarkusLanger** [reply]
@FranckyC Yes, you can see it in my screenshot of the App Catalog

**Reply by MarkusLanger** [reply]
The PnP Modern Search is deployed in the site app catalog. Should it be also in the global App Catalog?

**Reply by FranckyC** [reply]
Sorry I didn't pay attention :/. Any errors in the console/network tab when loading the property pane? Did you try in a private window? Also since 4.7.0, a layout type needs to be set in the layout definition `ILayoutDefinition`.

**Reply by MarkusLanger** [reply]
No errors in console or network tab. Private window does not work either. A layout type is defined. https://github.com/MarkusLanger/pnp-search-custom-layout/blob/check/src/libraries/infoScreenExtensibility/InfoScreenExtensibilityLibrary.ts#L18

**Reply by MarkusLanger** [reply]
@FranckyC Were you able to check my repository?

**Reply by FranckyC** [reply]
Yes and that was a bug in the latest version. Not a big thing but enough to not load the layout in the property pane:/ Your code was OK. One thing: you can now add the `renderType` property to the layout definition to specificy Handlebars or AdaptiveCards.

**Reply by MarkusLanger** [reply]
@FranckyC Yeah, works fine with "@pnp/modern-search-extensibility@1.3.0". I get an error with version 1.3.15. I will create a new ticket for this.

**Reply by FranckyC** [reply]
No need to create a ticket for this, the `1.3.15` is not aimed to be used in your code (this is @wobba experimentation with SPFx 1.15.0 :D)

**Reply by MarkusLanger** [reply]
That explains the error :-) . But it is already displayed as the latest version.

![image](https://user-images.githubusercontent.com/12131918/178487928-31592119-533b-42ea-ae85-586114279b3a.png)
